### PR TITLE
Mysql Table are not returning valid values

### PR DESCRIPTION
**Pull request by agardou**
links https://github.com/Opla/backend/issues/27

### CODE REVIEW

File: tests/database/mysql/mysqlTable.test.js [hunk 1]
```diff
@@ -166,26 +166,72 @@ describe("database/mysql/mysqlTable", () => {
...
         });
       });
 
       const items = await table1.getItems();
       expect(items.map((i) => i.id)).toEqual(["it-2", "it-3", "it-1"]);
     });
   });
+
+  describe("collection size", () => {
+    it("size without query", async () => {
```
Comment (willdurand) [inline tests/database/mysql/mysqlTable.test.js]: what does that even mean? 😂

File: tests/database/mysql/mysqlTable.test.js [hunk 2]
```diff
@@ -166,26 +166,72 @@ describe("database/mysql/mysqlTable", () => {
...
+          order: item.order,
         });
       });
 
       const items = await table1.getItems();
       expect(items.map((i) => i.id)).toEqual(["it-2", "it-3", "it-1"]);
     });
   });
+
+  describe("collection size", () => {
```
Comment (willdurand) [inline tests/database/mysql/mysqlTable.test.js]: `describe("size", ...)`

File: tests/database/mysql/mysqlTable.test.js [hunk 3]
```diff
@@ -166,26 +166,72 @@ describe("database/mysql/mysqlTable", () => {
...
+      count = await table1.size();
+      expect(count).toEqual(1);
+
+      await table1.setItem(null, {
+        id: "id2",
+      });
+      count = await table1.size();
+      expect(count).toEqual(2);
+    });
+    it("size with query", async () => {
```
Comment (willdurand) [inline tests/database/mysql/mysqlTable.test.js]: add a blank line before the next `it` block

Comment (willdurand) [reply tests/database/mysql/mysqlTable.test.js]: and add a better explanation for this test case too

Comment (agardou) [reply tests/database/mysql/mysqlTable.test.js]: is this ok ?

File: tests/database/mysql/mysqlTable.test.js (hunk 2 above)
Comment (agardou) [reply tests/database/mysql/mysqlTable.test.js]: fixed

File: tests/database/mysql/mysqlTable.test.js (hunk 1 above)
Comment (agardou) [reply tests/database/mysql/mysqlTable.test.js]: updated

File: tests/database/mysql/mysqlTable.test.js [hunk 4]
```diff
@@ -166,26 +166,73 @@ describe("database/mysql/mysqlTable", () => {
...
         });
       });
 
       const items = await table1.getItems();
       expect(items.map((i) => i.id)).toEqual(["it-2", "it-3", "it-1"]);
     });
   });
+
+  describe("size", () => {
+    it("size without where statement", async () => {
```
Comment (willdurand) [inline tests/database/mysql/mysqlTable.test.js]: > it returns the number of results

File: tests/database/mysql/mysqlTable.test.js [hunk 5]
```diff
@@ -166,26 +166,73 @@ describe("database/mysql/mysqlTable", () => {
...
+      expect(count).toEqual(1);
+
+      await table1.setItem(null, {
+        id: "id2",
+      });
+      count = await table1.size();
+      expect(count).toEqual(2);
+    });
+
+    it("size with where statement", async () => {
```
Comment (willdurand) [inline tests/database/mysql/mysqlTable.test.js]: > it returns the number of results restricted by a WHERE clause

Comment (agardou) [reply tests/database/mysql/mysqlTable.test.js]: :-) fixed

File: tests/database/mysql/mysqlTable.test.js (hunk 4 above)
Comment (agardou) [reply tests/database/mysql/mysqlTable.test.js]: :-) fixed

File: tests/database/mysql/mysqlTable.test.js [hunk 6]
```diff
@@ -166,26 +166,73 @@ describe("database/mysql/mysqlTable", () => {
...
         });
       });
 
       const items = await table1.getItems();
       expect(items.map((i) => i.id)).toEqual(["it-2", "it-3", "it-1"]);
     });
   });
+
+  describe("size", () => {
+    it("it returns the number of results", async () => {
```
Comment (willdurand) [inline tests/database/mysql/mysqlTable.test.js]: ah well you already have `it`

Comment (agardou) [reply tests/database/mysql/mysqlTable.test.js]: ha ! ^^

**Comment by willdurand** [comment]
thanks

**Pull request merged**